### DISCUSSION TITLE
Toggle blur on the fly

### DISCUSSION THREAD

**Discussion by dhruvinsh** [wezterm/wezterm#4516] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows, Linux Wayland, macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

Hi, is it possible to implement on the fly blur changes with some keymap?

**Comment by nekowinston** [comment, marked as answer]
As an example:
```lua
local wezterm = require("wezterm")

local toggleBlur = wezterm.action_callback(function(window)
  if window:effective_config().window_background_opacity == 1 then
    window:set_config_overrides({ window_background_opacity = 0.8 })
  else
    window:set_config_overrides({ window_background_opacity = 1 })
  end
end)

return {
  keys = {
    {
      mods = 'CTRL',
      key = 'b',
      action = toggleBlur,
    },
  },
}
```

**Reply by dhruvinsh** [reply]
Thanks. Ahh i was missing action_callbacj. I will give this a try.
